### PR TITLE
Add support for bun

### PR DESCRIPTION
**Pull request by BaylorRae**
This replaces the use of Node/Yarn with [bun](https://bun.sh) when `bun.lockb` is present.

Thanks for the patterns you've set up in this action to make this change so easy!

Screenshot 2023-09-13 at 12 06 42 PM

### CODE REVIEW

**Comment by andyw8** [comment]
Thanks, I will take a look at this soon.

**Comment by BaylorRae** [comment]
Thanks @andyw8! There's no rush, I was reviewing open PRs on mobile and accidentally closed this one 🤦‍♂️

**Comment by andyw8** [comment]
I see Rails will ship with support for this: https://tomaszs2.medium.com/rails-7-1-beta-eases-docker-releases-and-boosts-javascript-with-bun-7a9418f6def8

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -144,24 +164,47 @@ jobs:
         with:
           files: "package.json"
 
+      - name: Check for bun.lockb existence
+        id: bun_lock_check
+        uses: andstor/file-existence-action@v2
+        with:
+          files: "bun.lockb"
+
+      - name: Check if Sorbet is in use
```
Comment (andyw8) [inline .github/workflows/verify.yml]: Looks like this was accidental, bad rebase?

File: .github/workflows/verify.yml [hunk 2]
```diff
@@ -144,24 +164,47 @@ jobs:
...
           node-version-file: ${{ inputs.node-version-file }}
           cache: "yarn"
 
       - name: Install Node packages
-        if: steps.package_json_check.outputs.files_exists == 'true'
+        if: steps.package_json_check.outputs.files_exists == 'true' && steps.bun_lock_check.outputs.files_exists == 'false'
         run: |
           yarn install --pure-lockfile
 
+      - name: Set up Bun
```
Comment (andyw8) [inline .github/workflows/verify.yml:191]: Once this is merged, I think I'll restructure things so there is just a `Set up JavaScript` step which handles Yarn or Bun conditionally. Then there won't be so many skipped steps listed in the GitHub Actions output.

File: .github/workflows/verify.yml [hunk 3]
```diff
@@ -90,10 +105,15 @@ jobs:
...
       - name: Install Node packages
-        if: steps.package_json_check.outputs.files_exists == 'true'
+        if: steps.package_json_check.outputs.files_exists == 'true' && steps.bun_lock_check.outputs.files_exists == 'false'
         run: |
           yarn install --pure-lockfile
 
+      - name: Install Bun packages
+        if: steps.package_json_check.outputs.files_exists == 'true' && steps.bun_lock_check.outputs.files_exists == 'true'
+        run: |
+          bun install --pure-lockfile
```
Comment (andyw8) [inline .github/workflows/verify.yml]: Is `--pure-lockfile` supported? I can't find it mentioned in the docs. (`bun` doesn't seem to complain if you pass it invalid arguments).

Comment (BaylorRae) [reply .github/workflows/verify.yml]: Yes, it's in the docs for `bun install` https://bun.sh/docs/cli/install.

Here's a direct link to the docs for this option.

https://github.com/oven-sh/bun/blob/bbc70d2b73c80630a48cab5a24d8e00b07032e12/docs/install/index.md?plain=1#L52-L55

File: .github/workflows/verify.yml (hunk 1 above)
Comment (BaylorRae) [reply .github/workflows/verify.yml]: Good catch 😬 I've rebased my commits to remove the merge bubble and correct this error!

File: .github/workflows/verify.yml (hunk 3 above)
Comment (andyw8) [reply .github/workflows/verify.yml]: Looks like it's `--frozen-lockfile` rather than `--pure-lockfile`?

**Pull request merged**